src/java/fr/paris/lutece/portal/service/workflow/WorkflowService.java

```diff
@@ -409,6 +409,26 @@ public String getDisplayDocumentHistory( int nIdResource, String strResourceType
         }
     }
 
+    /**
+     * Get the confirmation of action processing 
+     * 
+     * @param nIdResource
+     *            the resource id
+     * @param strResourceType
+     *            the resource type
+     * @param nIdAction
+     *            the action id
+     * @param locale
+     *            the locale
+     * @param strTemplate
+     *            The template
+     * @return the confirmation
+     */
+    public String getDisplayProcessActionConfirmation( int nIdResource, String strResourceType, int nIdAction, Locale locale, String strTemplate )
```

Comment:
Is the purpose of this method to bring back the last action history?
should it not be named in this sense ?

Reply:
the purpose of this method is to bring back a confirmation of the processing of an action triggered by the user to be displayed on his screen

Reply:
This solution is interesting, but I think we can go further, for example to be able to trace back the cases of transaction rollback during the processing of actions.

Currently the doProcessAction() method does not return anything, it could return the return of the actions executed (an action can automatically trigger another one).

We could use an object of type RessourceHistory for this, this would allow to have something like:

List<ResourceHistory> actionHistoryResourceList = workflowService.doProcessAction( ... );

We would use the resourceHistory that are created to be inserted into the database, and in the case of a transaction rollback, we could create a specific ResourceHistory.

We can complete the ResourceHistory class with an int "status" attribute that would have the values ​​of an ENUM (1:SUCCESS, -1:FAILURE) for example (by assigning SUCCESS by default for upward compatibility)

Would it be possible to continue on these bases and propose a PR in this direction?

Reply:
I thought of a similar solution at first place, but when we trigger the processing of a workflow action we can be in one of two scenarios. 
In the first scenario we call the doProcessAction() method and in the second scenario we call the doSaveTasksForm() method. The latter however, returns errors if there are any.
With this piece of information, do you think the proposed solution is still relevant ?

Reply:
ok, i see,  doSaveTasksForm() should only return error messages...
we could create another method signature for doSaveTasksForm() with another argument, a ResourceHistory object reference, to be valued by doProcessAction() ?

Reply:
Hello,
the requested change has been made.
if it's OK for you, then the workflow core library version have to be upgraded in lutece-core to take into account the change made to this dependancy.